src/geom-transform.cpp

```diff
@@ -86,44 +89,52 @@ bool geom_transform_area_t::set_param(char const *name, lua_State *lua_state)
 bool geom_transform_area_t::is_compatible_with(
     table_column_type geom_type) const noexcept
 {
-    if (m_multi) {
-        return geom_type == table_column_type::multipolygon ||
-               geom_type == table_column_type::geometry;
-    }
-
     return geom_type == table_column_type::polygon ||
+           geom_type == table_column_type::multipolygon ||
            geom_type == table_column_type::geometry;
 }
 
 geom::osmium_builder_t::wkbs_t
 geom_transform_area_t::run(geom::osmium_builder_t *builder,
+                           table_column_type target_geom_type,
                            osmium::Way *way) const
 {
     assert(builder);
     assert(way);
 
+    geom::osmium_builder_t::wkbs_t result;
+
     if (!way->is_closed()) {
-        return {};
+        return result;
     }
 
-    geom::osmium_builder_t::wkbs_t result;
     result.push_back(builder->get_wkb_polygon(*way));
 
     if (result.front().empty()) {
         result.clear();
+    } else if (target_geom_type == table_column_type::multipolygon) {
+        builder->wrap_in_multipolygon(&result);
     }
 
     return result;
 }
 
 geom::osmium_builder_t::wkbs_t
 geom_transform_area_t::run(geom::osmium_builder_t *builder,
+                           table_column_type target_geom_type,
                            osmium::Relation const &relation,
                            osmium::memory::Buffer const &buffer) const
 {
     assert(builder);
 
-    return builder->get_wkb_multipolygon(relation, buffer, m_multi);
+    auto create_multigeoms = geom::create_multi_t::never;
+    if (target_geom_type == table_column_type::multipolygon) {
+        create_multigeoms = geom::create_multi_t::always;
+    } else if (m_multi) {
+        create_multigeoms = geom::create_multi_t::optional;
+    }
```

Comment:
If I read this right, then the `multi` flag is ignored, when the type is MULTIPOLYGON. Makes perfect sense but it might be worth documenting (and possibly testing, to document that this was the intended behaviour ;).

Reply:
After some more thought, this wasn't the right choice. The `multi` setting should be independent of the target geometry type. The new code implements this that way.